public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -102,6 +102,17 @@ export function transformV2(
     (df) => isHeatmapResult(df, request)
   );
 
+  // this works around the fact that we only get back frame.name with le buckets when legendFormat == {{le}}...which is not the default
+  heatmapResults.forEach((df) => {
+    if (df.name == null) {
+      let le = df.fields.find((f) => f.name === 'Value')?.labels?.le;
+
+      if (le != null) {
```

Comment:
Could we simplify this to `if(le)` instead of `le != null`?

Reply:
i think this would have unintended consequences if `le` was `0`

Reply:
trying this comment again, cause i confused myself for a second :smile: 

i think `if(le)` would miss setting the frame name if `le` was `0`

Reply:
the label values are all strings, so we'll never actually have `0`...and `'0'` will still be okay.

but yeah, i generally hate coalescing to truthy/falsey, even in `if ()`. `== null` is more chars but covers only null and undefined, without having to think through what else might fail.

i changed it for you, though, @gtk-grafana; anything for you :heart: